src/EFCore.PG/Query/NpgsqlSqlExpressionFactory.cs

```diff
@@ -279,34 +279,29 @@ SqlExpression ApplyTypeMappingOnSqlBinary(SqlBinaryExpression binary, Relational
                 var leftType = left.Type.UnwrapNullableType();
                 var rightType = right.Type.UnwrapNullableType();
 
-                if (binary.OperatorType == ExpressionType.Add)
+                // Note that we apply the given type mapping from above to the left operand (which has the same CLR type as
+                // the binary expression's)
+
+                // DateTime + TimeSpan => DateTime
+                // DateTimeOffset + TimeSpan => DateTimeOffset
+                if (rightType == typeof(TimeSpan) && (
+                        leftType == typeof(DateTime) ||
+                        leftType == typeof(DateTimeOffset)) ||
+                    rightType.FullName == "NodaTime.Period" && (
+                        leftType.FullName == "NodaTime.LocalDateTime" ||
+                        leftType.FullName == "NodaTime.LocalDate" ||
+                        leftType.FullName == "NodaTime.LocalTime") ||
+                    rightType.FullName == "NodaTime.Duration" && (
+                        leftType.FullName == "NodaTime.Instant" ||
+                        leftType.FullName == "NodaTime.ZonedDateTime"))
                 {
-                    // Note that we apply the given type mapping from above to the left operand (which has the same CLR type as
-                    // the binary expression's)
-
-                    // DateTime + TimeSpan => DateTime
-                    // DateTimeOffset + TimeSpan => DateTimeOffset
-                    if (rightType == typeof(TimeSpan) && (
-                            leftType == typeof(DateTime) ||
-                            leftType == typeof(DateTimeOffset)) ||
-                        rightType.FullName == "NodaTime.Period" && (
-                            leftType.FullName == "NodaTime.LocalDateTime" ||
-                            leftType.FullName == "NodaTime.LocalDate" ||
-                            leftType.FullName == "NodaTime.LocalTime") ||
-                        rightType.FullName == "NodaTime.Duration" && (
-                            leftType.FullName == "NodaTime.Instant" ||
-                            leftType.FullName == "NodaTime.ZonedDateTime"))
-                    {
-                        var newLeft = ApplyTypeMapping(left, typeMapping);
-                        var newRight = ApplyDefaultTypeMapping(right);
-                        return new SqlBinaryExpression(binary.OperatorType, newLeft, newRight, binary.Type, newLeft.TypeMapping);
-                    }
+                    var newLeft = ApplyTypeMapping(left, typeMapping);
+                    var newRight = ApplyDefaultTypeMapping(right);
+                    return new SqlBinaryExpression(binary.OperatorType, newLeft, newRight, binary.Type, newLeft.TypeMapping);
                 }
 
                 if (binary.OperatorType == ExpressionType.Subtract)
                 {
-                    var inferredTypeMapping = typeMapping ?? ExpressionExtensions.InferTypeMapping(left, right);
```

Comment:
Eh, just happened to move this one closer to where it's used.
Can revert this, it's a pointless "fix".

Reply:
Nah, it's fine...